### ISSUE TITLE
IfStatementAssignment thrown incorrectly

### ISSUE THREAD

**Issue by leewillis77** [phpmd/phpmd#661]
- PHPMD version: 2.7.0
- PHP Version: 7.1.30
- Installation type: composer
- Operating System / Distribution & Version: MacOS Mojave

### Current Behavior
Sample code below throws two IfStatementAssignment errors.
```php
<?php

function bar($thing, $ids)
{
    if ($thing->status === 'CREATED') {
        $ids->each(function ($id) use ($thing) {
            $allocation         = find($id);
            $allocation->status = 'paid';
        });
    }
}
```
The errors reported are:
test.php:3	Avoid assigning values to variables in if clauses and the like (line '13', column '7').
test.php:3	Avoid assigning values to variables in if clauses and the like (line '13', column '8').

Accounting for the incorrect line/column numbers (see #655), this is complaining about the two assignments within the closure.

### Expected Behavior
No IfStatementAssignment errors should be thrown as is the case if the wrapping function is removed, e.g. the code below throws no IfStatementAssignment errors.

```php
<?php

if ($thing->status === 'CREATED') {
    $ids->each(function ($id) use ($thing) {
        $allocation         = find($id);
        $allocation->status = 'paid';
    });
}
```

### Steps To Reproduce:
See examples above.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
I tried to get a minimum code chunk to reproduce the bug and get it with:
```php
function bar()
{
    if (true) {
        array_filter(function ($foo) {
            $bar = $foo;
            echo $bar;
        });
    }
}
```

Minimum tokens needed seem to be: a statement, a function/method call with a closure parameter, an assignation inside the closure.

**Comment by wernerdweight** [comment]
Is there any estimation on when 2.7.1 (fixes this) could be released?

**Comment by kylekatarnls** [comment]
We just starting to work on it: #668

**Comment by kylekatarnls** [comment]
Thanks for your help @leewillis77 it will be fixed in the next release, you can test it with the `dev-master` version.

**Comment by leewillis77** [comment]
While the test cases posted here seem no longer to throw the error, I'm still seeing this error generated with my *actual* code.

I'm going to see if I can produce an alternative test case, would you prefer to re-open this issue, or would you prefer me to open a fresh one?

**Comment by leewillis77** [comment]
Examples below:

## Does NOT throw the error
```php
<?php

function bar($ids)
{
    if ('CREATED' === 'CREATED') {
        $ids->each(function () {
            $allocation = find(1);
            $allocation->status = 'paid';
            $allocation->potato = 'round';
        });
        return true;
    }
}
```

## Throws an error, but should not
```php
<?php

function bar($ids, $allocation)
{
    if ('CREATED' === 'CREATED') {
        $ids->each(function () use ($allocation) {
            $allocation->status = 'paid';
            $allocation->potato = 'round';
        });
        return true;
    }
}
```

## Cut-down version of real code that throws error, but should not
```php
<?php

namespace App\Services;

use App\Models\Allocation;

class PayoutService
{
    public function payoutAllocations($allocationIds)
    {
        if ('CREATED' === 'CREATED') {
            $allocationIds->each(function ($allocationId) use ($payout) {
                $allocation                     = Allocation::find($allocationId);
                $allocation->status             = 'paid';
            });
        }
    }
}
```

**Comment by kylekatarnls** [comment]
I found the actual problem. I will fix it.

**Comment by tvbeek** [comment]
@leewillis77 with thanks to @kylekatarnls there is a new fix on master. Can you verify it?

**Comment by leewillis77** [comment]
Perfect - that no longer throws those errors on my test code, or the real code that triggered this initially. Thanks!
